### ISSUE TITLE
Add static analysis for all files in repository

### ISSUE THREAD

**Issue by mkordas** [Viridovix/show-me-the-code-github#19]
Checkstyle is able to perform basic analysis in non-Java files.

The following things can be enforced over entire repository:
- no tab characters used
- every file ends with newline
- no unnecessary consecutive lines
- no trailing whitespace characters at the end of lines
- files are not too long

Gradle is convenient tool to run the analysis. Fast feedback about violations can be achieved using TravisCI (#7).

Autogenerated files should be excluded from analysis.

**Comment by nejdaw** [comment]
We are not sure about the stack. Why are we adding Checkstyle? What if we will use sbt, or Grunt for JavaScript?

**Comment by nejdaw** [comment]
It would be good first to comment on #5 and #6 . Then we can make issues to add tools

**Comment by mkordas** [comment]
I don't want to impose any build tool. The only purpose of Gradle (actually Gradle Wrapper) would be to execute Checkstyle (or other static analysis tools in future) using `gradlew check`. If we decided to use sbt or Grunt, then they can serve as main build tools/dependency managers and invocation of `gradlew check` could be just one of their stages. What do you think?

**Comment by nejdaw** [comment]
I agree that static analysis will be essential for us. But I would leave the technical details to the moment when we decide on technology we will use. There might be better way to perform SA in other environments (although I admit that we will probably end up using checkstyle anyway). 

In any case, this Issue could have more generic title for the time being such as `Add static analysis for all files in repository`. We could also add information that it depends on other issues being addressed first.

**Comment by mkordas** [comment]
Thanks for suggestion, issue renamed. I'll propose PR with Gradle as this is super-easy, but I'm open for future replacement with another technology.

**Comment by nejdaw** [comment]
@mkordas since we have agreed to use ES6 in #6 I have suggestion. How about setting up [_CodeClimate_](https://codeclimate.com/) for our project? It does perform basing checks for JavaScript and rates the project in general.

Could you investigate whether it is possible to use it against ES6 (instead of ES5)?

If yes, then we would need to do:
- set up _CodeClimate_ to our project
- configure it
- change description for this issue
- possible decouple it from _Travis_ task

**Comment by mkordas** [comment]
@dawidnejman 
CodeClimate is already enabled (done in #10) and it [supports ES6](https://codeclimate.com/changelog/5474f0326956800dea007ed1), but there are no files to be analyzed yet - it only recognizes `.js` or `.css` files. The only purpose of this task is to perform very simple analysis of miscellaneous files in our project - for example that in JSON or YML spaces are used for indentation instead of tabs.

**Comment by nejdaw** [comment]
@mkordas How about using [_CodeClimate checkstyle engine_](https://github.com/sivakumar-kailasam/codeclimate-checkstyle) ? As I understand, we would not need _gradle_ wrapper to run _checsktyle_ checks.

**Comment by mkordas** [comment]
I'll look into it, but is it possible to easily run analysis locally before pushing code? Gradle is super convenient here, as it does not require any setup, it's just `gradlew check`.

**Comment by nejdaw** [comment]
Instead of using _Checkstyle_ we could use  [_JSHint_](http://jshint.com/) on local environment, which looks to be the tool of choice for JS projects.. What do you think? The only thing I am not sure is the ES6 support (although I have seen on blog that they are adding it at the moment - so it might be ready). Since we will have node project maybe this should be the way to go?

**Comment by mkordas** [comment]
Sure, but this would be separate issue, feel free to create it. This one is about very simple rules for all files including `.yml`, '.md', '.html' and '.json'. JSHint cannot analyze these.
